### PR TITLE
ENH: Add itk.xarray_from_image and itk.image_from_xarray

### PR DESCRIPTION
**Pull request by thewtex**
These functions convert an itk.Image to and from an xarray.DataArray. The
xarray.DataArray preserves spatial metadata while processing in Python.
xarray enables parallel computing with Dask, machine learning, and
modern seralization with Zarr.

These interfaces are complete but experimental. They are available for
testing but are subject to possible future changes and refinements.

### CODE REVIEW

File: Wrapping/Generators/Python/Tests/extras.py [hunk 1]
```diff
@@ -147,189 +147,246 @@ def custom_callback(name, progress):
...
-
-# test reading image series with `itk.imread()` and check that dimension is
-# not increased if last dimension is 1.
-image_series = itk.imread([image_series3d_filename, image_series3d_filename])
-assert image_series.GetImageDimension() == 3
-
-# pipeline, auto_pipeline and templated class are tested in other files
-
-# BridgeNumPy
+# # test mesh read / write
```
Comment (dzenanz) [inline Wrapping/Generators/Python/Tests/extras.py]: Why was this code commented out? It does not seem related to xarray.

Comment (thewtex) [reply Wrapping/Generators/Python/Tests/extras.py]: Good catch -- I will re-enable.

Comment (thewtex) [reply Wrapping/Generators/Python/Tests/extras.py]: Looks like @hjmjohnson caught it and fixed it -- thanks, Hans!

Comment (dzenanz) [reply Wrapping/Generators/Python/Tests/extras.py]: I don't see the fix.

Comment (thewtex) [reply Wrapping/Generators/Python/Tests/extras.py]: I was looking at the wrong file :-P -- I will patch it.
